src/client/Microsoft.Identity.Client/Internal/AuthCodeFetcher.cs

```diff
@@ -0,0 +1,264 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Instance;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.TelemetryCore.Internal;
+using Microsoft.Identity.Client.TelemetryCore.Internal.Events;
+using Microsoft.Identity.Client.UI;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.Internal
+{
+    /// <summary>
+    /// Responsible for getting an auth code
+    /// </summary>
+    internal class AuthCodeFetcher
```

Comment:
This class is much cleaner that the old InteractiveRequest - 1 ctor, 1 public method and all members are readonly

Reply:
There is no need to validate the logic here as it was copied - just the public surface